default/python/common/timers.py

```diff
@@ -0,0 +1,87 @@
+from time import time
+
+
+class Timer(object):
+    def __init__(self, timer_name):
+        """
+        Creates timer. Timer name is name that will be in logs.
+
+        """
+        self.timer_name = timer_name
+        self.start_time = None
+        self.end_time = None
+        self.section_name = None
+        self.timers = []
+        self.log_name = None
+
+    def stop(self, section_name=''):
+        """
+        Stop timer if running. Specify section_name if want to override its name.
+        """
+        if self.start_time:
+            self.end_time = time()
+            section_name = section_name or self.section_name
+            self.timers.append((section_name, (self.end_time - self.start_time) * 1000.0))
+        self.start_time = None
+        self.section_name = None
+
+    def start(self, section_name):
+        """
+        Stop prev timer if present and starts new.
+        """
+        self.stop()
+        self.section_name = section_name
+        self.start_time = time()
+
+    def _print_timer_table(self, time_table):
+        """
+        Print a header and a footer and fill it with the timing results from time_table
+        """
+        if not self.timers:
+            return
+        max_header = max(len(x[0]) for x in self.timers)
```

Comment:
Add  TODO  to use this https://github.com/freeorion/freeorion/blob/master/default/python/common/print_utils.py#L116